core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala

```diff
@@ -278,6 +278,12 @@ private[deploy] class SparkSubmitArguments(args: Seq[String], env: Map[String, S
     if (pyFiles != null && !isPython) {
       SparkSubmit.printErrorAndExit("--py-files given but primary resource is not a Python script")
     }
+    if (totalExecutorCores != null && executorCores != null
+        && (totalExecutorCores.toInt % executorCores.toInt) != 0) {
```

Comment:
Minor, but I think the parentheses are in the wrong place. They should go around the expression, or probably just be removed. You might avoid duplicating the mod expression for clarity.